stablehlo/dialect/StablehloOps.td

```diff
@@ -1867,10 +1867,17 @@ def StableHLO_CollectivePermuteOp: StableHLO_Op<"collective_permute",
 
   let arguments = (ins
     HLO_Tensor:$operand,
-    I64ElementsAttr:$source_target_pairs
+    I64ElementsAttr:$source_target_pairs,
+    OptionalAttr<StableHLO_ChannelHandle>:$channel_handle
   );
   let results = (outs HLO_Tensor);
   let hasVerifier = 1;
+  // channel_handle is only used for the SPMD partitioner, so we add a
+  // simplified builder method for convenience.
+  let builders = [
+    OpBuilder<(ins
+      "::mlir::Type":$result_type, "::mlir::Value":$operand,
+      "::mlir::DenseIntElementsAttr":$source_target_pairs)>];
 }
 
 def StableHLO_ConvolutionOp : StableHLO_Op<"convolution", [Pure]> {
```

Comment:
This change will need a test in `ops_stablehlo.mlir` (to test that `channel_handle` is supported) and in `ops_stablehlo_roundtrip.mlir` (to test that it can be serialized / deserialized).

Reply:
Added additional test in `ops_stablehlo_roundtrip.mlir` leaving the original test as is (both with and without optional parameter code should be roundtrippable), added a non failing test in `ops_stablehlo.mlir`